azkaban-common/src/main/java/azkaban/jobExecutor/AbstractProcessJob.java

```diff
@@ -161,15 +162,21 @@ public Props getJobGeneratedProperties() {
    * @return {tmpPropFile, outputPropFile}
    */
   public File[] initPropsFiles() {
-    // Create properties file with additionally all input generated properties.
-    final File[] files = new File[2];
-    files[0] = createFlattenedPropsFile(this._cwd);
+    // Create job properties file with additionally all input generated properties.
+    final File[] files = new File[3];
+    files[0] = createFlattenedPropsFile(this.jobProps, this._cwd, "_job_props_");
 
     this.jobProps.put(ENV_PREFIX + JOB_PROP_ENV, files[0].getAbsolutePath());
     this.jobProps.put(ENV_PREFIX + JOB_NAME_ENV, getId());
 
     files[1] = createOutputPropsFile(getId(), this._cwd);
     this.jobProps.put(ENV_PREFIX + JOB_OUTPUT_PROP_FILE, files[1].getAbsolutePath());
+
+    // Create job's system properties file.
+    // TODO : ProcessJob.java:254
+    files[2] = createFlattenedPropsFile(this.sysProps, this._cwd, "_job_sys_props_");
+    this.jobProps.put(ENV_PREFIX + JOB_SYS_PROP_ENV, files[2].getAbsolutePath());
```

Comment:
How do we know we are not exposing any application(Azkaban) level sensitive configuration by exposing system properties?

Reply:
These properties consist of only jobtype plugin properties.

Reply:
I believe that properties in "commonprivate.properties" file which are properties set by Azkaban, not visible to users and common to all jobtypes are included in _sysProps_. In addition private.properties file which are specific to jobtypes are also in sysProps according to documentation https://azkaban.readthedocs.io/en/latest/getStarted.html#property-overrides. By listing them in a file inside the flow's execution directory we are making them accesible to users. Users have write permissions on that folder.

Reply:
Let us discuss this in person.